pkg/pipelines/service.go

```diff
@@ -143,6 +148,49 @@ func serviceResources(m *config.Manifest, fs afero.Fs, p *AddServiceParameters)
 	return res.Merge(built, files), nil
 }
 
+func inheritBindings(cicdEnv, env *config.Environment, svc *config.Service) ([]string, error) {
+	var bindings []string
+	if svc.SourceURL == "" {
+		return []string{}, nil
+	}
+	// inherit pipelines from env
+	if env.Pipelines != nil {
+		bindings = env.Pipelines.Integration.Bindings
+	}
+	repo, err := scm.NewRepository(svc.SourceURL)
+	if err != nil {
+		return []string{}, err
+	}
+	_, svcBinding := repo.CreatePRBinding(cicdEnv.Name)
+	// remove bindings of other repo types to avoid param name collision
+	bindings = extractBindings(bindings)
+
+	// add the pipelines for the service repo type
+	bindings = append(bindings, svcBinding)
+
+	return bindings, nil
+}
+
+// extract all the bindings except service bindings
+func extractBindings(bindings []string) []string {
+	var svcBindings []string
+	for i := range bindings {
+		if !isSpecial(bindings[i]) {
+			svcBindings = append(svcBindings, bindings[i])
+		}
+	}
+	return svcBindings
+}
+
+func isSpecial(binding string) bool {
```

Comment:
isSpecial is not very descriptive function name.    I'd call it something like "isRepositoryBinding" or isSCMBinding().   Also, I'd move this function to scm package and hide the GetAllBindings().

Reply:
Updated